metrics/firebase/functions/lib/main.dart

```diff
@@ -61,47 +137,93 @@ int _getSuccessfulBuildDuration(BuildData buildData) {
 /// from the given [dateTime].
 DateTime _getUtcDay(DateTime dateTime) => dateTime.toUtc().date;
 
+/// Creates an instance of the [BuildDayStatusField] based on
+/// the given [buildStatus] and [incrementCount].
+BuildDayStatusField _createBuildDayStatusField({
```

Comment:
Maybe it would be great to move this logic to a separate `factory` class? Would it make our code a bit more testable and clean?